src/memory/server/MemoryServer.ts

```diff
@@ -99,9 +110,44 @@ export class MemoryServer {
         const session = vscode.debug.activeDebugSession;
         if (session) {
             try {
-                const result: MemoryContents = await session.customRequest(
-                    'cdt-gdb-adapter/Memory',
-                    request.args
+                if (request.args.child === undefined) {
+                    const result: MemoryContents = await session.customRequest(
+                        'cdt-gdb-adapter/Memory',
+                        request.args
+                    );
+                    this.sendResponse(panel, request, {
+                        result,
+                    });
+                } else {
+                    const result: MemoryContents = await session.customRequest(
+                        'cdt-amalgamator/Memory',
+                        request.args
+                    );
+                    this.sendResponse(panel, request, {
+                        result,
+                    });
+                }
```

Comment:
I don't think we need the whole thing copy-pasted, just the one parameter is different.
```suggestion
                    const result: MemoryContents = await session.customRequest(
                        request.args.child === undefined ? 'cdt-gdb-adapter/Memory' : 'cdt-amalgamator/Memory',
                        request.args
                    );
                    this.sendResponse(panel, request, {
                        result,
                    });
```

Reply:
I have updated it.